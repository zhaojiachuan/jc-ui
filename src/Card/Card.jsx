import React from 'react'
import { Divider } from 'antd'


const index = ({ extra, children, title, borderColor, width, borderRadius, height, margin }) => {
    let boxStyle;
    let titleStyle;
    let contentStyle;
    let titleText;
    let extraStyle;
    boxStyle = {
        border: `2px solid ${borderColor}`,
        width: `${width}`,
        height: `${height}`,
        borderRadius: `${borderRadius}`,
        margin: `${margin}`
    };
    titleStyle = {
        height: '3vh',
        lineHeight: '6vh',
        paddingLeft: '1vw',
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'space-between',
        paddingRight: '1vw'
    };
    contentStyle = {
        paddingLeft: '1vw'
    };
    titleText = {
        fontSize: 'larger',
    };
    extraStyle = {

    }
    return (
        <div style={boxStyle}>
            <div style={titleStyle}>
                <div style={titleText}>{title}</div>
                <div style={extraStyle}>{extra}</div>
            </div>
            <Divider />
            <div style={contentStyle}>{children}</div>
        </div>
    )
}

index.defaultProps = {
    borderColor: "#f0f0f0",
    width: '23vw',
    height: '26vh',
    borderRadius: '0px'
}

export default index