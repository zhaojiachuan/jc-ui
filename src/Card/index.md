---
title: Card 卡片
nav:
  title: 组件
  path: /componments
group:
  title: 数据展示
  path: /basic
  order: 3
---

# Card 卡片
通用卡片容器。
<br/>

## 何时使用
最基础的卡片容器，可承载文字、列表、图片、段落，常用于后台概览页面。

## 代码演示

### 基础使用

<code src='./demos/demo1.jsx'></code>
<br/>
设置extra来确定是否自定义右侧内容，支持一切格式

### 自定义卡片样式 
<code src='./demos/demo2.jsx'></code>
<br/>
用户可以通过style属性自行设定样式，也可以通过borderColor、borderRadius等属性快速设定
### Button

| 参数  |                            说明                             |                 类型                  | 默认值 | 版本  |
| :---: | :---------------------------------------------------------: | :-----------------------------------: | :----: | :---: |
| async |           异步操作 onClick 执行时自动添加 loading           |                boolean                | false  |   -   |
| Role  | 角色权限 具体看 [Roles 组件](/hera-ui/business/roles#roles) | string \| (string \| (()=>boolean))[] |   -    | 2.2.3 |

兼容其他 antd Button 属性
