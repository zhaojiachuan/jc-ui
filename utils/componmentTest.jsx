import React from 'react'
import { render } from '@testing-library/react';

export default function componmentTest (Componment, ComponmentName, ComponmentProps) {
    // 传递三个参数 组件，组件名称，组件属性
    describe(`component ${ComponmentName}`, () => {
        it(`renders correctly`, () => {
            // 第一条测试用例
            const wrapper = render(<Componment {...ComponmentProps} />);
            // 快照测试，判断能不能正常渲染
            expect(wrapper.container).toMatchSnapshot();
        });
        it('component could be updated and unmounted without errors', () => {
            // 第二条测试用例
            const { rerender, unmount } = render(<Componment {...ComponmentProps} />);
            // 判断是否能rerender和卸载
            expect(() => {
                rerender(<Componment {...ComponmentProps} />);
                unmount();
            }).not.toThrow();
        });
    });
}