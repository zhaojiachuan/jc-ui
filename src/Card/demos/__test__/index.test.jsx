import componmentTest from "../../../../utils/componmentTest";
import Card from '../../index';

componmentTest(Card, 'Card', { children: 'card' });