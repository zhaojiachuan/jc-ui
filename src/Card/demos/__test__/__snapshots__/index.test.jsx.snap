// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Card renders correctly 1`] = `
<div>
  <div
    style="border: 2px solid #f0f0f0; width: 23vw; height: 26vh; border-radius: 0px;"
  >
    <div
      style="height: 3vh; line-height: 6vh; padding-left: 1vw; display: flex; flex-direction: row; justify-content: space-between; padding-right: 1vw;"
    >
      <div
        style="font-size: larger;"
      />
      <div />
    </div>
    <div
      class="ant-divider css-dev-only-do-not-override-mxhywb ant-divider-horizontal"
      role="separator"
    />
    <div
      style="padding-left: 1vw;"
    >
      card
    </div>
  </div>
</div>
`;
